src/_algopy_testing/serialize.py

```diff
@@ -25,66 +25,71 @@ def identity(i: _T) -> _T:
     return i
 
 
-def get_native_to_arc4_serializer(typ: type[_T]) -> _Serializer:  # type: ignore[type-arg] # noqa: PLR0911
+def get_native_to_arc4_serializer(typ: type[_T]) -> _Serializer:  # type: ignore[type-arg]  # noqa: PLR0911, PLR0912
     from _algopy_testing import arc4
     from _algopy_testing.models import Account
     from _algopy_testing.primitives import BigUInt, Bytes, ImmutableArray, String
     from _algopy_testing.protocols import UInt64Backed
 
-    if issubclass(typ, arc4._ABIEncoded):
+    origin_typ = typing.get_origin(typ)
```

Comment:
Nitpick: I think this should be declared closer to the usage or perhaps even inlined as it is used in only one spot.